src/vivarium/framework/values/manager.py

```diff
@@ -513,19 +513,15 @@ def register_rate_producer(
         self,
         rate_name: str,
         source: Callable[..., Any],
-        # TODO [MIC-5452]: all calls should have a component
```

Comment:
register_rate_producer is weird b/c it's an older interface metho (i.e. I didn't want to remove the old-style requires_* args, but it calls the new register_attribute_producer method which did have those stripped out.

I just updated it to require only `required_resources` and luckily I didn't need to fix much.